Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -26,10 +26,14 @@ public record ProcessedProgram(Program Program, Action<VCGen, Implementation, Ve
 
   public class ExecutionEngine : IDisposable
   {
+    private static readonly CustomStackSizePoolTaskScheduler largeThreadScheduler = CustomStackSizePoolTaskScheduler.Create(16 * 1024 * 1024, Environment.ProcessorCount);
```

Comment:
Worth a quick comment stating why we need such a deep stack, maybe point to https://github.com/boogie-org/boogie/issues/711

Reply:
Done